src/solidity-frontend/solidity_convert.cpp

```diff
@@ -3477,4 +3484,163 @@ void solidity_convertert::convert_type_expr(
 
   else
     solidity_gen_typecast(ns, src_expr, dest_type);
+}
+
+static inline void static_lifetime_init(const contextt &context, codet &dest)
+{
+  dest = code_blockt();
+
+  // call designated "initialization" functions
+  context.foreach_operand_in_order([&dest](const symbolt &s) {
+    if(s.type.initialization() && s.type.is_code())
+    {
+      code_function_callt function_call;
+      function_call.function() = symbol_expr(s);
+      dest.move_to_operands(function_call);
+    }
+  });
+}
+
+/*
+  verify the contract as a whole
+*/
+bool solidity_convertert::move_functions_to_main(
+  const std::string &contractName)
+{
+  // return if "function" is set or "contract" is unset
+  if(
+    !config.options.get_option("function").empty() ||
+    config.options.get_option("contract").empty())
+    return false;
+
+  // return if it's not the target contract
+  if(contractName != config.options.get_option("contract"))
+    return false;
+
+  codet init_code, body_code;
+  static_lifetime_init(context, body_code);
+  static_lifetime_init(context, init_code);
+
+  //body_code.make_block();
+  init_code.make_block();
+
+  // get contract symbol "tag-contract"
+
+  const std::string id = prefix + contractName;
+  if(context.find_symbol(id) == nullptr)
+    return true;
+
+  const symbolt &contract = *context.find_symbol(id);
+
+  // a contract should be a struct
+  assert(contract.type.is_struct());
+
+  // 1. call constructor()
+  if(context.find_symbol(contractName) == nullptr)
+    return true;
+  const symbolt &constructor = *context.find_symbol(contractName);
+  code_function_callt call;
+  call.location() = constructor.location;
+  call.function() = symbol_expr(constructor);
+  init_code.move_to_operands(call);
+
+  // 2. do while(nondet_uint())
+
+  // 3. do if(nondet_uint()) then func()
+  const struct_typet::componentst &methods =
+    to_struct_type(contract.type).methods();
+  for(const auto &method : methods)
+  {
+    // extract method(function)
+    const std::string func_id = method.identifier().as_string();
+    // skip constructor
+    if(func_id == contractName)
+      continue;
+
+    if(context.find_symbol(func_id) == nullptr)
+      return true;
+
+    const symbolt &func = *context.find_symbol(func_id);
+
+    // construct ifthenelse
+
+    // guard: nondet_uint()
```

Comment:
why do you need nondet_uint? Why not nondet_ushort?

Reply:
My mistake. Will fix.

Reply:
My mistake. Will fix.